standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/Warehouse.java

```diff
@@ -363,8 +366,14 @@ public Path getDefaultTablePath(Database db, String tableName, boolean isExterna
     } else {
       dbPath = getDatabaseManagedPath(db);
     }
-    return getDnsPath(
-        new Path(dbPath, MetaStoreUtils.encodeTableName(tableName.toLowerCase())));
+    if (!isExternal && tableName.matches("(.*)" + V_TABLE_PATTERN)) {
```

Comment:
nit: Maybe a comment, or a different String value, since this is not a `tableName`. This is `tablePath`? Strings are always recreated anyway, so there is not performance loss, but I think it would be easier to understand the code

Reply:
sorry, don't follow,  tableName is actually a table name, like `acid_tbl.00005`. dbPath contains an absolute path.